File: 0231-power-of-two/0231-power-of-two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        long long pow = 2;

        if(n == 1 || n == 2) return true; 
        for(int i = 2; i <= n; i++){
            pow *= 2;
            if(pow == n) return true;
            if(pow > n) return false;
        }

        return false;
    }
};

File: 0231-power-of-two/0231-power-of-two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n == 1: return True
        pow = 1
        for i in range(1,n+1):
            pow *= 2
            if pow == n: 
                return True
            if pow > n:
                return False
        return False


        